src/Sentry.Unity/SentryMonoBehaviour.cs

```diff
@@ -112,3 +113,26 @@ private void Awake()
         DontDestroyOnLoad(gameObject);
     }
 }
+
+/// <summary>
+/// A MonoBehaviour that captures screenshots
+/// </summary>
+public partial class SentryMonoBehaviour
+{
+    // Todo: keep track of event ID to not capture double/more than once per frame
+
+    public void CaptureScreenshotForEvent(SentryUnityOptions options, SentryId eventId)
+    {
+        StartCoroutine(CaptureScreenshot(options, eventId));
```

Comment:
This method returns before the coroutine starts/ends, right? how do we know the screenshot will be taken before the event is created?

Reply:
The ScreenshotProcessor forwards the call to the MonoBehaviour that starts the coroutine for the current EndOfFrame.